activemodel/lib/active_model/type/strict_integer.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  module Type
+    class StrictInteger < Integer  # :nodoc:
+      private
+
+        def cast_value(value)
+          case value
+          when true then 1
+          when false then 0
+          when /\A[\d\.]+\Z/ then value.to_i
```

Comment:
This regular expression is incorrect. It does not match `-1`, a valid integer. It does match `1.0.0`, which is not a valid integer.